lib/puppet/transaction/report.rb

```diff
@@ -240,6 +240,46 @@ def name
     host
   end
 
+  # Provide a summary of the changes
+  def display_changes
```

Comment:
Based on the name `display_changes`, I'd expect this method to display the changes, or rename the method to `get_changes`, and have the caller print them like you're doing.